src/content_script/input_tools.js

```diff
@@ -0,0 +1,237 @@
+let SLSettings = {};
+
+const InputTools = {
+  isLoading: false,
+
+  // store tracked input elements
+  trackedElements: [],
+
+  init(target) {
+    InputTools.queryEmailInputAndApply(target, (element) => {
+      if (!InputTools.isValidEmailInput(element)) {
+        return;
+      }
+
+      // ignore if this elements has already been tracked
+      const i = InputTools.trackedElements.indexOf(element);
+      if (i === -1) {
+        InputTools.trackedElements.push(element);
+        InputTools.addSLButtonToInput(element);
+      }
+    });
+  },
+
+  destroy(target) {
+    InputTools.queryEmailInputAndApply(target, (element) => {
+      // remove element from tracking list
+      const i = InputTools.trackedElements.indexOf(element);
+      if (i !== -1) {
+        InputTools.trackedElements.splice(i, 1);
+      }
+    });
+  },
+
+  queryEmailInputAndApply(target, actionFunction) {
+    if (!target.querySelectorAll) return;
```

Comment:
how this condition can happen?

Reply:
I actually ran into a problem when `MutationObservation` returns something other than DOM Node. Still don't know the cause